nav2_dwb_controller/dwb_core/src/dwb_local_planner.cpp

```diff
@@ -469,57 +477,62 @@ DWBLocalPlanner::transformGlobalPlan(
           ControllerTFError("Unable to transform robot pose into global plan's frame");
   }
 
-  // we'll discard points on the plan that are outside the local costmap
+  // Discard points on the plan that are outside the local costmap
   nav2_costmap_2d::Costmap2D * costmap = costmap_ros_->getCostmap();
   double dist_threshold = std::max(costmap->getSizeInCellsX(), costmap->getSizeInCellsY()) *
     costmap->getResolution() / 2.0;
 
-
-  // If prune_plan is enabled (it is by default) then we want to restrict the
-  // plan to distances within that range as well.
-  double prune_dist = prune_distance_;
-
-  // Set the maximum distance we'll include points before getting to the part
-  // of the path where the robot is located (the start of the plan). Basically,
-  // these are the points the robot has already passed.
-  double transform_start_threshold;
-  if (prune_plan_) {
-    transform_start_threshold = std::min(dist_threshold, prune_dist);
-  } else {
-    transform_start_threshold = dist_threshold;
-  }
-
-  // Set the maximum distance we'll include points after the part of the part of
-  // the plan near the robot (the end of the plan). This determines the amount
-  // of the plan passed on to the critics
-  double transform_end_threshold;
-  if (shorten_transformed_plan_) {
-    transform_end_threshold = std::min(dist_threshold, prune_dist);
-  } else {
-    transform_end_threshold = dist_threshold;
-  }
-
-  // Find the first pose in the global plan that's further than prune distance
-  // from the robot using integrated distance
-  auto prune_point = nav2_util::geometry_utils::first_after_integrated_distance(
```

Comment:
This is to help with paths with overlapping segments to select the closest one to the actual robot. WIthout this, you're not guaranteed to pick the first of 2 similarly positioned points on a "crossing" of the path -- instead relying on a coin flip about which is "absolutely" closer by small floating point error.

Reply:
Hmm this is a complexe issue. For now, I don't see any way to find with certitude the closest point to the robot. We could bound the research range using the closest point found at the previous iteration, and the maximum speed provided as parameter (max_vel_x/y) multiplied by the controller frequency. Still we cannot exclude that the robot moves faster than this.